controller/replicator.go

```diff
@@ -312,6 +312,94 @@ func (r *replicator) Snapshot(name string, userCreated bool, created string) err
 	return nil
 }
 
+func (r *replicator) GetLastSnapshot() (string, error) {
+	retErrorLock := sync.Mutex{}
+	retError := &BackendError{
+		Errors: map[string]error{},
+	}
+	replicaChains := make(map[string][]string)
+	wg := sync.WaitGroup{}
+	var success int
+
+	for addr, backend := range r.backends {
+		if backend.mode == types.RW {
+			wg.Add(1)
+			go func(address string, backend types.Backend) {
+				if chain, err := backend.GetReplicaChain(); err != nil {
+					logrus.Infof("failed taking snapshot at %s with err %v", addr, err)
+					retErrorLock.Lock()
+					retError.Errors[address] = err
+					retErrorLock.Unlock()
+				} else {
+					retErrorLock.Lock()
+					replicaChains[addr] = chain
+					success = success + 1
+					retErrorLock.Unlock()
+				}
+				wg.Done()
+			}(addr, backend.backend)
+		} else {
+			logrus.Infof("not taking snapshot at %s in err mode", addr)
+		}
+	}
+	wg.Wait()
+
+	if len(retError.Errors) != 0 {
+		return "", retError
+	}
+	var snapName string
+	for _, chain := range replicaChains {
```

Comment:
should we make sure that chains are also same at all the places?
and also should check for replicationFactor also matching with len(replicaChains)?

Reply:
will be adding a check to match Replication factor with replicaChains.
Matching the chains is already performed when a replica is converted to RW mode in verifyRebuildReplica().